### PR TITLE
Makefile: testnet targets wrapping connect_to_testnet.sh

### PR DESCRIPTION
**Pull request by stefantalpalaru**
- also moved the binary into the "build" directory

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,6 +43,22 @@ clean_eth2_network_simulation_files:
 eth2_network_simulation: | beacon_node validator_keygen clean_eth2_network_simulation_files
 	SKIP_BUILDS=1 GIT_ROOT="$$PWD" BUILD_OUTPUTS_DIR="./build" tests/simulation/start.sh
 
+clean-testnet0:
```
Comment (zah) [inline Makefile]: It's not appropriate for the default target to reset your database before each connection.
The client will starting syncing from scratch then, which will take longer and longer.

Comment (Swader) [reply Makefile]: We discussed this and concluded that for the sake of easier debugging and bug reporting on the often-restarted testnets, a good default is clearing the cache and redownloading.

**Comment by stefantalpalaru** [comment]
The idea is that you recommend `make testnet0` when you don't want to debug old cache data, which is most of the time, and `make testnet0-nocleaning` after you've explained that it reuses the existing cache.

File: Makefile (hunk 1 above)
Comment (stefantalpalaru) [reply Makefile]: The idea is that you recommend `make testnet0` when you don't want to debug old cache data, which is most of the time, and `make testnet0-nocleaning` after you've explained that it reuses the existing cache.

**Pull request merged**